### PR TITLE
Dependency Bumps and Apache httpclient5 migration

### PR DESCRIPTION
**Pull request by The-Funk**
#### Pull Request Description

-updated all dependencies to latest
-migrated apache httpclient to version 5

---

#### Future Release Comment
This upgrades the version of the Apache httpclient used by the library to version 5 for incoming support of asynchronous/non-blocking ADX queries.

**Breaking Changes:**
There are 2 test failures total.

1. createExceptionFromBlankErrorMessage() in UtilitiesTest in kusto-data module
2. E2ETest in kusto-ingest

There is a TODO in the code for the max connections parameter. It does not appear that version 5 of the Apache http client supports a max connections parameter.

**Features:**
Everything is up to date and new Apache client appears to work

**Fixes:**
NA

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,32 +32,32 @@
...
+        <httpclient5.version>5.1.3</httpclient5.version>
+        <httpcore5.version>5.1.3</httpcore5.version>
+        <json.version>20220320</json.version>
+        <fasterxml.jackson.core.version>2.13.3</fasterxml.jackson.core.version>
         <azure-storage.version>8.6.6</azure-storage.version>
-        <azure-identity.version>1.3.7</azure-identity.version> <!-- Do not use 1.4.1, as it has a showstopping bug for us. Any version that includes https://github.com/Azure/azure-sdk-for-java/pull/25580 is OK -->
-        <azure-core.version>1.21.0</azure-core.version>
-        <reactor-core.version>3.4.10</reactor-core.version>
+        <azure-identity.version>1.5.2</azure-identity.version> <!-- Do not use 1.4.1, as it has a showstopping bug for us. Any version that includes https://github.com/Azure/azure-sdk-for-java/pull/25580 is OK -->
+        <azure-core.version>1.29.1</azure-core.version>
```
Comment (ohadbitt) [inline pom.xml:51]: Please don't touch azure core versions

**Review by ohadbitt — CHANGES_REQUESTED** [review]
This simply doesnt work now with trying to execute simple commands ( tried to create a table)
The returned response is not a json

**Comment by ohadbitt** [comment]
doesn't seem like you changed something to make it work - please try to invoke any command using this

**Comment by ohadbitt** [comment]
> I'll give it a go. How do you want me to run tests? I've just been doing mvn test and everything appears okay with that minus the two tests I listed above. You want me to build a jar, import it in a project, connect it to my own ADX instance and then manually test?
> 
> I noticed a similar JSON issue in 3.1.2. For some reason JSON isn't working properly for me in that version. However, 3.1.1 works fine.

All tests should pass
If e2e passes thats fine with fixing what i said.

File: data/src/main/java/com/microsoft/azure/kusto/data/Utils.java [hunk 2]
```diff
@@ -53,30 +47,26 @@ private Utils() {
...
                     case HttpStatus.SC_OK:
-                        return responseContent;
+                        return response.toString();
                     case HttpStatus.SC_TOO_MANY_REQUESTS:
                         throw new ThrottleException(urlStr);
                     default:
-                        throw createExceptionFromResponse(urlStr, response, null, responseContent);
+                        throw createExceptionFromResponse(urlStr, response, null, response.toString());
                 }
             }
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/Utils.java:71]:
```
 try (CloseableHttpResponse response = httpClient.execute(request)) {
                if (response != null) {
                    String responseContent = EntityUtils.toString(response.getEntity());

                    switch (response.getCode()) {
                        case HttpStatus.SC_OK:
                            return responseContent;
                        case HttpStatus.SC_TOO_MANY_REQUESTS:
                            throw new ThrottleException(urlStr);
                        default:
                            throw createExceptionFromResponse(urlStr, response, null, responseContent);
                    }
                }
            }
```


You need to use Entity to get the actual content of the response, response.toString() just gives the status line.

Fix this and the tests will pass.

**Comment by The-Funk** [comment]
Oh my goodness. I can't believe I forgot to change that back. I added response.getEntity().toString(), if you'd like to use EntityUtils to do that conversion I can switch back to EntityUtils.

**Comment by AsafMah** [comment]
@The-Funk still doesn't work for me, entity.toString() gives me "Wrapper [Wrapper [[Content-Type: application/json; charset=UTF-8,Content-Encoding: gzip,Chunked: true]]]"

You can just use the code I gave.

Also please resolve the merge conflicts if you can.

**Comment by The-Funk** [comment]
> @The-Funk still doesn't work for me, entity.toString() gives me "Wrapper [Wrapper [[Content-Type: application/json; charset=UTF-8,Content-Encoding: gzip,Chunked: true]]]"
> 
> You can just use the code I gave.
> 
> Also please resolve the merge conflicts if you can.

Sounds good. I'll try to work on this today, sorry for the delay.

**Comment by The-Funk** [comment]
@AsafMah a lot on the go today so I might have rushed those merge conflict fixes. Hopefully third time is the charm. Haven't tried E2ETest yet.

**Comment by The-Funk** [comment]
Nevermind, I think I accidentally kept your changes and deleted mine. Not my day. :(

**Comment by AsafMah** [comment]
There seemed to be some merge complications,
So I re-did it myself, and solved the todos/fixme.

The PR is now in #257, so I'm closing this one.

**Pull request closed without merging**